### PR TITLE
[Kotlin Migration] create personal account - part 1

### PR DESCRIPTION
**Pull request by mejdoo**
## What's new in this PR?

Layout implementation with custom styles for create personal account.
https://app.zeplin.io/project/5e256b5d43c3b4295ef1a75b/screen/5e2597e2887fd66f0f13daf7
https://wearezeta.atlassian.net/browse/AN-6682

device-2020-02-24-145407

#### APK
[Download build #1358](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1358/artifact/build/artifact/wire-dev-PR2638-1358.apk)
[Download build #1364](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1364/artifact/build/artifact/wire-dev-PR2638-1364.apk)

### CODE REVIEW

File: app/src/main/res-migration/layout/fragment_create_personal_account.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        android:layout_height="wrap_content"
+        android:gravity="center_horizontal"
+        android:text="@string/new_reg__phone_signup__create_account"
+        android:textColor="@color/outer_space"
+        android:textSize="@dimen/create_personal_account_title_text_view_size"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/topGuideline"
+        />
+
```
Comment (BradleyWire) [inline app/src/main/res-migration/layout/fragment_create_personal_account.xml]: Very minor but too much whitespace here

File: app/src/main/res-migration/layout/fragment_create_personal_account_with_phone.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            style="@style/EditText.Authentication"
+            />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <androidx.appcompat.widget.AppCompatImageButton
+        android:id="@+id/confirmationButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        style="@style/Button.Authentication"
```
Comment (BradleyWire) [inline app/src/main/res-migration/layout/fragment_create_personal_account_with_phone.xml]: Format this layout file as the style attribute should live at the top

File: app/src/main/res-migration/layout/fragment_create_personal_account_with_email.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="@string/profile__email__placeholder"
+            android:inputType="textEmailAddress"
+            style="@style/EditText.Authentication"
+            />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <androidx.appcompat.widget.AppCompatImageButton
```
Comment (BradleyWire) [inline app/src/main/res-migration/layout/fragment_create_personal_account_with_email.xml]: format this layout

File: app/src/main/res-migration/layout/fragment_create_personal_account.xml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/createPersonalAccountTitleTextView"
+        app:layout_constraintVertical_bias="0.146">
+
+        <com.google.android.material.tabs.TabLayout
+            android:id="@+id/createPersonalAccountTabLayout"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            style="@style/TabLayout"
```
Comment (BradleyWire) [inline app/src/main/res-migration/layout/fragment_create_personal_account.xml]: format this xml

File: app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountFragment.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+class CreatePersonalAccountFragment : Fragment(R.layout.fragment_create_personal_account) {
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val tabTitles = listOf(getString(R.string.profile__email__placeholder),
+            getString(R.string.pref_account_phone_title))
+
+        createPersonalAccountViewPager.adapter =
+            CreatePersonalAccountViewPagerAdapter(requireActivity().supportFragmentManager, tabTitles)
```
Comment (gizemb) [inline app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountFragment.kt]: aren't you adding those fragments to CreatePersonalAccountFragment? Why are you passing activity's fragmentManager here? It should be this fragment's (child)fragmentManager imo 🤔

File: app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountWithEmailFragment.kt [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import androidx.fragment.app.Fragment
+import com.waz.zclient.R
+import kotlinx.android.synthetic.main.fragment_create_personal_account_with_email.*
+
+class CreatePersonalAccountWithEmailFragment : Fragment(R.layout.fragment_create_personal_account_with_email) {
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+    }
+
```
Comment (gizemb) [inline app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountWithEmailFragment.kt]: you can delete this

File: app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountWithPhoneFragment.kt [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import android.view.View
+import androidx.fragment.app.Fragment
+import com.waz.zclient.R
+
+class CreatePersonalAccountWithPhoneFragment : Fragment(R.layout.fragment_create_personal_account_with_phone) {
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+    }
+
```
Comment (gizemb) [inline app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountWithPhoneFragment.kt]: same

File: app/src/main/res-migration/layout/fragment_create_personal_account.xml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        android:layout_height="wrap_content"
+        android:gravity="center_horizontal"
+        android:text="@string/new_reg__phone_signup__create_account"
+        android:textColor="@color/outer_space"
+        android:textSize="@dimen/create_personal_account_title_text_view_size"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/topGuideline" />
+
+    <androidx.viewpager.widget.ViewPager
```
Comment (gizemb) [inline app/src/main/res-migration/layout/fragment_create_personal_account.xml]: androidx has ViewPager2 now. let's use the newest widgets.

File: app/src/main/res-migration/layout/fragment_create_personal_account_with_email.xml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/create_personal_account_text_input_layout_margin_top"
+        android:background="@color/white"
+        app:hintEnabled="false"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <androidx.appcompat.widget.AppCompatEditText
```
Comment (gizemb) [inline app/src/main/res-migration/layout/fragment_create_personal_account_with_email.xml]: TextInputEdittext?

File: app/src/main/res-migration/values/styles_tablayout.xml [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="TabLayout" parent="Widget.Design.TabLayout">
```
Comment (gizemb) [inline app/src/main/res-migration/values/styles_tablayout.xml]: I think you should extract those features to a feature-specific tablayout as this design is not the same throughout the app

File: app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountWithEmailFragment.kt (hunk 6 above)
Comment (mejdoo) [reply app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountWithEmailFragment.kt]: I will need later to handle click listener of the confirmation button

File: app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountWithPhoneFragment.kt (hunk 7 above)
Comment (mejdoo) [reply app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountWithPhoneFragment.kt]: same

File: app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountFragment.kt (hunk 5 above)
Comment (mejdoo) [reply app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountFragment.kt]: Good catch ! 👍

File: app/src/main/res-migration/layout/fragment_create_personal_account_with_email.xml (hunk 9 above)
Comment (mejdoo) [reply app/src/main/res-migration/layout/fragment_create_personal_account_with_email.xml]: done

File: app/src/main/res-migration/values/styles_tablayout.xml (hunk 10 above)
Comment (mejdoo) [reply app/src/main/res-migration/values/styles_tablayout.xml]: done

File: app/src/main/res-migration/layout/fragment_create_personal_account.xml (hunk 8 above)
Comment (mejdoo) [reply app/src/main/res-migration/layout/fragment_create_personal_account.xml]: done 👍

File: app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountFragment.kt (hunk 5 above)
Comment (mejdoo) [reply app/src/main/kotlin/com/waz/zclient/auth/registration/personal/CreatePersonalAccountFragment.kt]: I changed this later again due to the migration to viewPager2. It requires now an instance of the fragmentActivity

**Comment by mejdoo** [comment]
build again

**Comment by mejdoo** [comment]
build again

**Pull request merged**